backend/file/file.go

```diff
@@ -7,20 +7,18 @@ import (
 	"path/filepath"
 
 	"github.com/go-yaml/yaml"
-	"github.com/heetch/confita/backend"
 	"github.com/pkg/errors"
 )
 
 // Backend that loads a configuration from a file.
 // It supports json and yaml formats.
 type Backend struct {
-	path        string
-	unmarshaler backend.ValueUnmarshaler
-	name        string
+	path string
+	name string
 }
 
 // NewBackend creates a configuration loader that loads from a file.
-// The content will get decoded based on the file extension and cached in the backend.
```

Comment:
Why did you remove the cache "feature" ?

Reply:
The interface is not the same anymore. `ValueUnmarshaler` -> `Unmarshaler` meaning the whole file is unmarshaled and not single values which required using a cache (otherwhile for each field we would have to reopen the file)